### PR TITLE
Handle flags with multiple inputs without the flag before each input

### PR DESCRIPTION
**Pull request by idandrd**
## What
So far the parser supported flags with multiple inputs only if the flag shows before each input,
so this worked: ```MyCli mycmd -i foo -i bar```
but this didn't: ```MyCli mycmd -i foo bar```

## Why
When passing FS paths with wildcards, the paths are parsed by the terminal before they reach the oclif code. so if I type: ```MyCli readFiles -i ./files/*.txt``` the oclif will actually get ```MyCli readFiles -i ./files/a.txt ./files/b.txt ./files/c.txt``` and the command will fail.
I needed a way to allow my users to select multiple files in one command, so this functionality was needed.

## How
I kept a copy of the last flag parsed. if the next string in argv is not a flag, and the last flag parsed is accepting multiple values, the string will be added as a value for this flag.

### CODE REVIEW

File: src/parse.ts [hunk 1]
```diff
@@ -47,8 +47,10 @@ export class Parser<T extends ParserInput, TFlags extends OutputFlags<T['flags']
   private readonly raw: ParsingToken[] = []
   private readonly booleanFlags: { [k: string]: Flags.IBooleanFlag<any> }
   private readonly context: any
+  private _currentFlag: Flags.IOptionFlag<any> | null
```
Comment (jdxcode) [inline src/parse.ts]: don't use an underscore here

**Comment by jdxcode** [comment]
I enabled circle to run on forked builds so it should build on the next push. I'm looking into why appveyor is failing, but that is unrelated to this PR. You submitted your last push just before I put a note in about using `null` (not sure if you saw that).

If you fix the null thing it should trigger a circle build and we can merge this

File: src/parse.ts (hunk 1 above)
Comment (jdxcode) [reply src/parse.ts]: in other words, this should be `private currentFlag?: Flags.IOptionFlag<any>`

File: test/parse.test.ts [hunk 2]
```diff
@@ -277,6 +277,33 @@ See more help with --help`)
...
+    it('flag multiple without flag in the middle', () => {
+      const out = parse(['--foo', './a.txt', './b.txt', './c.txt', '--hello', 'world'], {
+        flags: {foo: flags.string({multiple: true}), hello: flags.string()},
+      })
+      expect(out.flags).to.deep.include({foo: ['./a.txt', './b.txt', './c.txt']})
+      expect(out.flags).to.deep.include({hello: 'world'})
+    })
+
+    it('flag multiple with arguments', () => {
+      const out = parse(['--foo', './a.txt', './b.txt', './c.txt', '--', '15'], {
```
[the remark this answers is not in the document]
Comment (idandrd) [reply test/parse.test.ts]: than you :)

**Comment by idandrd** [comment]
Thank you very much!
BTW, thanks a lot for creating Oclif, it really helps a lot in our startup.

**Pull request merged**